src/pages/filter/ui/query-builder/right-refiner-column/func-filter/func-filter.tsx

```diff
@@ -0,0 +1,77 @@
+import { ReactElement } from 'react'
+
+import { FuncStepTypesEnum } from '@core/enum/func-step-types-enum'
+import {
+  ICompoundHetArgs,
+  ICompoundRequestArgs,
+  ICustomInheritanceModeArgs,
+  IGeneRegionArgs,
+  IInheritanceModeArgs,
+  TFuncArgs,
+} from '@service-providers/common/common.interface'
+import { CompoundHetView } from './components/compound-het-view'
+import { CompoundRequestView } from './components/compound-request-view'
+import { CustomInheritanceModeView } from './components/custom-inheritance-mode-view'
+import { GeneRegionView } from './components/gene-region-view'
+import { InheritanceModeView } from './components/inheritance-mode-view'
+
+interface IFuncFilterProps {
+  filterId: string
+  filterName: string
+  isFilterActive: boolean
+  filterContent: string[]
+  filterExpression: TFuncArgs
+}
+
+export const FuncFilter = ({
+  filterId,
+  filterName,
+  isFilterActive,
+  filterContent,
+  filterExpression,
+}: IFuncFilterProps): ReactElement => (
+  <>
```

Comment:
Maybe with switch it will be more readable, as I see, you return only one element depend on Enum value, so why not?

Reply:
Probably its a matter of taste, but I don't like using a switch statements